src/stores/__tests__/bots.spec.js

```diff
@@ -0,0 +1,83 @@
+import { setActivePinia, createPinia } from 'pinia'
+
+import {
+	getConversationBots,
+	disableBotForConversation,
+	enableBotForConversation,
+} from '../../services/botsService.js'
+import { generateOCSResponse } from '../../test-helpers.js'
+import { useBotsStore } from '../bots.js'
+
+jest.mock('../../services/botsService', () => ({
+	getConversationBots: jest.fn(),
+	disableBotForConversation: jest.fn(),
+	enableBotForConversation: jest.fn(),
+}))
+
+describe('botsStore', () => {
+	beforeEach(async () => {
+		setActivePinia(createPinia())
+	})
+
+	it('returns an empty array if bots are not available', async () => {
+		const botsStore = useBotsStore()
+		expect(botsStore.getConversationBots('empty-token')).toEqual([])
+	})
+
+	it('receives a response from server, stores and updates bots', async () => {
+		const token = 'TOKEN'
+		const botsStore = useBotsStore()
+
+		// First response
+		const oldBots = [
+			{ id: 1, name: 'Dummy bot 1', state: 0 },
+			{ id: 2, name: 'Dummy bot 2', state: 1 },
+		]
+		let response = generateOCSResponse({ payload: oldBots })
+		getConversationBots.mockResolvedValueOnce(response)
+		await botsStore.loadConversationBots(token)
+
+		expect(getConversationBots).toHaveBeenCalledWith(token)
+		expect(botsStore.getConversationBots(token)).toEqual(oldBots)
+
+		// New response with updated bots list
+		const newBot = { id: 3, name: 'Dummy bot 3', state: 2 }
+		response = generateOCSResponse({ payload: [...oldBots, newBot] })
+		getConversationBots.mockResolvedValueOnce(response)
+		await botsStore.loadConversationBots(token)
+
+		expect(getConversationBots).toHaveBeenCalledWith(token)
+		expect(botsStore.getConversationBots(token)).toEqual([...oldBots, newBot])
+	})
+
+	it('toggles bots state', async () => {
+		const token = 'TOKEN'
+		const botsStore = useBotsStore()
+
+		const bots = [
+			{ id: 1, name: 'Dummy bot 1', state: 0 },
+			{ id: 2, name: 'Dummy bot 2', state: 1 },
+		]
+		const response = generateOCSResponse({ payload: bots })
+		getConversationBots.mockResolvedValueOnce(response)
+		await botsStore.loadConversationBots(token)
+
+		// Enable bot with id: 1
+		const enabledBot = { ...bots[0], state: 1 }
+		const responseEnabled = generateOCSResponse({ payload: enabledBot })
+		enableBotForConversation.mockResolvedValueOnce(responseEnabled)
+
+		await botsStore.toggleBotState(token, bots[0])
+		expect(enableBotForConversation).toHaveBeenCalledWith('TOKEN', bots[0].id)
+		expect(botsStore.getConversationBots('TOKEN')).toEqual([enabledBot, bots[1]])
+
+		// Disable bot with id: 2
+		const disabledBot = { ...bots[1], state: 0 }
+		const responseDisabled = generateOCSResponse({ payload: disabledBot })
+		disableBotForConversation.mockResolvedValueOnce(responseDisabled)
+
+		await botsStore.toggleBotState(token, bots[1])
+		expect(disableBotForConversation).toHaveBeenCalledWith('TOKEN', bots[1].id)
+		expect(botsStore.getConversationBots('TOKEN')).toEqual([enabledBot, disabledBot])
+	})
```

Comment:
```suggestion
	it('toggles bots state', async () => {
		const token = 'TOKEN'
		const botsStore = useBotsStore()

		const bots = [
			{ id: 1, name: 'Dummy bot 1', state: 0 },
			{ id: 2, name: 'Dummy bot 2', state: 1 },
		]
		const response = generateOCSResponse({ payload: bots })
		getConversationBots.mockResolvedValueOnce(response)
		await botsStore.loadConversationBots(token)

		// Enable bot with id: 1
		const enabledBot = { ...bots[0], state: 1 }
		const responseEnabled = generateOCSResponse({ payload: enabledBot })
		enableBotForConversation.mockResolvedValueOnce(responseEnabled)
		await botsStore.toggleBotState(token, bots[0])

		// Disable bot with id: 2
		const disabledBot = { ...bots[1], state: 0 }
		const responseDisabled = generateOCSResponse({ payload: disabledBot })
		disableBotForConversation.mockResolvedValueOnce(responseDisabled)
		await botsStore.toggleBotState(token, bots[1])

		expect(enableBotForConversation).toHaveBeenCalledWith(token , bots[0].id)		
		expect(disableBotForConversation).toHaveBeenCalledWith(token , bots[1].id)
		expect(botsStore.getConversationBots(token)).toEqual([enabledBot, disabledBot])
	})
```

Reply:
In unit tests it is a good practice to follow the AAA pattern:
1. Arrange - prepare the environment for the test, set up the initial state and constants:
   - Setup initial store state
2. Act - make the action being tested
3. Assert - do the actual check

Then it is clear what exactly in what case is tested. And it saves from creating tests that are actually a sequence of several tests.

It is different from the e2e test where we test not units but long user scenarios with many actions and complex initial setup.